src/target/riscv/riscv-013.c

```diff
@@ -1362,6 +1363,18 @@ static void deinit_target(struct target *target)
 	info->version_specific = NULL;
 }
 
+static int set_haltgroup(struct target *target, bool *supported)
+{
+	uint32_t write = set_field(DMI_DMCS2_HGWRITE, DMI_DMCS2_HALTGROUP, target->smp);
+	if (dmi_write(target, DMI_DMCS2, write) != ERROR_OK)
```

Comment:
I don't see a read/modify/write, just a write, but I assume this does not clobber any non-haltgroup fields of DMCS2 (or if it does potentially clobber then there's no bad effect from that)?

Reply:
So far writing 0 to the other fields of dmcs2 is what we want. If more fields are added in the future, we might have to update this code to use those features.

Reply:
OK, yeah, looks good to me.